lib/Controller/Display.php

```diff
@@ -1836,4 +1838,46 @@ public function moveCms($displayId)
             throw new InvalidArgumentException(__('Invalid Two Factor Authentication Code'), 'twoFactorCode');
         }
     }
+
+    public function addViaCodeForm()
+    {
+        $this->getState()->template = 'display-form-addViaCode';
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function addViaCode()
+    {
+        $user_code = $this->getSanitizer()->getString('user_code');
+        $cmsAddress = (new HttpsDetect())->getUrl();
+        $cmsKey = $this->getConfig()->getSetting('SERVER_KEY');
+
+        if ($user_code == '') {
+            throw new InvalidArgumentException('Code cannot be empty', 'code');
+        }
+
+        $guzzle = new Client();
+
+        try {
+            $request = $guzzle->request('POST', 'https://auth.signlicence.co.uk/addDetails',
```

Comment:
I think it would be sensible to add some comments here explaining what this URL is and what this does in general, so that people are not concerned about what is happening to their data.